### PR TITLE
Label offsets

### PR DESCRIPTION
**Pull request by karimnaaji**
Mostly useful for points for now, line offsets will behave weirdly (especially when tilting, rotating), will follow up on more advanced label line offsets once we decided of the right convention.

### CODE REVIEW

File: core/src/scene/drawRule.h [hunk 1]
```diff
@@ -13,17 +13,18 @@
...
 enum class StyleParamKey : uint8_t {
     none, order, extrude, color, width, cap, join, outline_color, outline_width, outline_cap, outline_join,
     font_family, font_weight, font_style, font_size, font_fill, font_stroke, font_stroke_color, font_stroke_width, font_uppercase,
-    visible, priority
+    visible, priority, offset
 };
 
 struct StyleParam {
-    using Value = variant<none_type, std::string, CapTypes, JoinTypes, Extrusion, int32_t, uint32_t, float, bool>;
+    using Value = variant<none_type, std::string, CapTypes, JoinTypes, Extrusion, Offset, int32_t, uint32_t, float, bool>;
```
Comment (blair1618) [inline core/src/scene/drawRule.h]: Might make sense to combine the `Extrusion` and `Offset` variations into a single `std::pair<float, float>` variation, to reduce the generated code size a bit.

File: core/resources/scene/scene.yaml [hunk 2]
```diff
@@ -104,10 +104,10 @@ layers:
                 width: .02
                 cap: round
                 color: white
-                outline:
-                    width: .01
-                    cap: round
-                    color: '#999'
+#outline:
```
Comment (tallytalwar) [inline core/resources/scene/scene.yaml]: Was this for debugging? We can put this back!

File: core/src/scene/sceneLoader.cpp [hunk 3]
```diff
@@ -64,7 +64,11 @@ std::string parseSequence(const Node& node) {
         try {
             sstream << val.as<float>() << ",";
         } catch (const BadConversion& e) {
-            logMsg("Error: Float value was expected for styleParam sequence value\n");
+            try {
+                sstream << val.as<std::string>() << ",";
+            } catch (const BadConversion& e) {
+                logMsg("Error: Float or Unit expected for styleParam sequence value\n");
```
Comment (tallytalwar) [inline core/src/scene/sceneLoader.cpp]: I think any yaml scalar value can be converted to string, so not sure if it makes sense to have a try catch for `val.as<std::string>()`

**Comment by tallytalwar** [comment]
Some trivial suggestions in comments, apart from that looks good.

File: core/src/scene/drawRule.cpp [hunk 4]
```diff
@@ -228,6 +234,41 @@ bool DrawRule::operator<(const DrawRule& _rhs) const {
...
+        return false;
+    }
+
+    std::string value = _value;
+
+    // replace all unit occurences
+    for (auto& unit : _allowedUnit) {
+        auto i = value.find(unit);
+        // TODO: conversion
+        while (i != std::string::npos) {
```
Comment (karimnaaji) [inline core/src/scene/drawRule.cpp]: Yes!

Comment (karimnaaji) [inline core/src/scene/drawRule.cpp]: This step is only replacement to have a clean string though, I added an early exit if this step fails.

File: core/src/scene/sceneLoader.cpp (hunk 3 above)
Comment (karimnaaji) [inline core/src/scene/sceneLoader.cpp]: But what if you put another sequence in this sequence?

Comment (tallytalwar) [inline core/src/scene/sceneLoader.cpp]: Thats true.

**Comment by tallytalwar** [comment]
This one is also good to merge, once rebased with master.

**Pull request merged**